File: course/section_2_two_pointers/code/lesson_28_k-diff-pairs-in-an-array.py
from typing import List


class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        if k == 0:
            freq = {}
            for num in nums:
                freq[num] = freq.get(num, 0) + 1
            count = 0
            for s in freq.values():
                if s > 1:
                    count += 1
            return count
        count = 0
        unique = set(nums)
        for num in unique:
            if num + k in unique:
                count += 1
        return count

File: course/section_2_two_pointers/code/lesson_28_k-diff-pairs-in-an-array.php
<?php


class Solution {

    function findPairs($nums, $k) {
        $count = 0;
        if ($k === 0) {
            $freq = [];
            foreach ($nums as $num) {
                $freq[$num] = ($freq[$num] ?? 0) + 1;
            }
            foreach ($freq as $_ => $f) {
                if ($f > 1) {
                    $count++;
                }
            }
            return $count;
        }
        $unique = array_flip($nums);
        foreach ($unique as $num => $_) {
            if (isset($unique[$num + $k])) {
                $count++;
            }
        }
        return $count;
    }
}
